tests/functional/WebIfcApi.spec.ts

```diff
@@ -1,37 +1,454 @@
+                {
+                    type: 1,
+                    value: ''
+                }
+            ]
+        })
+        let project: any = ifcApi.GetLine(emptyFileModelID, 1);
+        expect(project.Description.value).toEqual("foo");
+    })
+    test('Can modify a line with a rawLineData', () => {
+        ifcApi.WriteRawLineData(emptyFileModelID, {
+            ID: 1,
+            type: 4251960020,
+            arguments: [{
+                    type: 1,
+                    value: '0cqv$41Df7ygSQzJTMBnvM'
+                },
+                {
+                    type: 1,
+                    value: ''
+                },
+                {
+                    type: 1,
+                    value: 'foobar'
+                },
+                null,
+                {
+                    type: 5,
+                    value: ''
+                },
+                {
+                    type: 1,
+                    value: ''
+                },
+                {
+                    type: 1,
+                    value: ''
+                },
+                {
+                    type: 1,
+                    value: ''
+                }
+            ]
+        })
+        let project: any = ifcApi.GetLine(emptyFileModelID, 1);
+        expect(project.Description.value).toBe("foobar");
+    })
+    test('can write a line by giving a line object', () => {
+        let projectBeforeWriting: any = ifcApi.GetAllLines(modelID);
+        let payload = {
+            expressID: 9999999,
+            type: 3856911033,
+            toType: 3856911033,
+            OwnerHistory: {
+                type: 5,
+                value: 41
+            },
+            Name: {
+                type: 1,
+                value: 'NZ-SHS beam:100x6.0SHS:823947'
+            },
+            Description: null,
+            ObjectType: {
+                type: 1,
+                value: 'NZ-SHS beam:100x6.0SHS'
+            },
+            ObjectPlacement: {
+                type: 5,
+                value: 9750
+            },
+            Representation: {
+                type: 5,
+                value: 9987
+            },
+            LongName: {
+                type: 1,
+                value: '823947'
+            },
+            CompositionType: undefined,
+            PredefinedType: undefined,
+            ElevationWithFlooring: undefined
+        }
+        ifcApi.WriteLine(modelID, payload);
+        let projectAfterWriting: any = ifcApi.GetAllLines(modelID);
+        expect(projectBeforeWriting.size() + 1).toEqual(projectAfterWriting.size());
+    })
+    test('can modify a line by giving a line object', () => {
+        let aSpace: any = ifcApi.GetLine(modelID, expressId);
+        aSpace.Name.value = "foo";
+        ifcApi.WriteLine(modelID, aSpace);
+        let aSpaceReloaded: any = ifcApi.GetLine(modelID, expressId);
+        expect(aSpaceReloaded.Name.value).toEqual("foo");
+    })
+    test('can Export File As IFC', () => {
```

Comment:
Hi Tony, I have a pull to test writing as well.  Looks like you have more coverage, which is great :)

Could you include an explicit round-trip including serialization?
  - read IFC
  - get property
  - change value
  - write property
  - write IFC
  - read new IFC
  - read property
  - expect new IFC's prop value == written prop value

here's mine:
https://github.com/pablo-mayrgundter/web-ifc/blob/write-fix/tests/functional/WebIfcApi.spec.ts#L36

Reply:
Hi Pablo,
All the methods are already tested so you can use them in any way. A WriteFile on web-ifc 0.0.36 fails so any scenario depending on writeFile will fail with this version. I don't think it is necessary at this point to make complexe scenario, except for documentation :) and documentation with test is the smart way.

However something is interresting with your test your tring to writeFile on a getItemProperties() result.

Maybe i am missing something what do you think ?